components/ExampleModal.js

```diff
@@ -0,0 +1,150 @@
+        CancelButton,
+        StyledTopNav,
+        ButtonWrapper,
+        StyledModal,
+        ContentContainer,
+        MainContent,
+        Title
+    } = internals;
+
+    useEffect(() => {
+
+        Animated.timing(marginAnim.current, {
+            toValue: keyboardSize,
+            ...Theme.animation.keyboard
+        }).start();
+    }, [keyboardSize]);
+
+    return (
+        <StyledModal
+            visible={isVisible}
+            backdropStyle={{ backgroundColor: 'rgba(56, 56, 56, 0.5)' }}
+            onBackdropPress={() => {
+
+                Keyboard.dismiss();
+                onDismissModal();
+            }}
+            {...props}
+        >
+
+            <Animated.ScrollView bounces={false} style={{ flex: 1, marginBottom: marginAnim.current }} keyboardShouldPersistTaps='handled'>
+                <StyledTopNav
+                    title={() => <Title>{title}</Title>}
+                    accessoryRight={<CloseButton onPress={onDismissModal} accessoryRight={<Icons color='white' name='x' size={20} />} />}
+                />
+                <ContentContainer>
+                    <MainContent>
+                        {children}
+                    </MainContent>
+                </ContentContainer>
+                <Divider />
+                <ButtonWrapper
+                    accessoryRight={
+                        <View style={{ flexDirection: 'row' }}>
+                            <CancelButton onPress={onPressCancel}>CANCEL</CancelButton>
+                            <AddButton disabled={shouldDisableSubmit} onPress={onPressSubmit} size='giant'>{affirmativeLabel}</AddButton>
+                        </View>
+                    }
+                />
+            </Animated.ScrollView>
+
+        </StyledModal>
+    );
+};
+
+internals.ContentContainer = Styled.View`
+    flex: 1;
+    flex-grow: 1;
+    background-color: white;
+    flex-direction: row;
+`;
+
+internals.MainContent = Styled.View`
+    flex: 1;
+    background-color:#eee;
+    padding: ${({ theme }) => theme.spacing(3,2)};
+`;
+
+internals.StyledButton = Styled(Button).attrs({ appearance: 'outline', size: 'giant' })``;
+
+internals.AddButton = Styled(internals.StyledButton).attrs({ appearance: 'filled' })``;
+
+internals.CancelButton = Styled(internals.StyledButton)`
+    margin-right: ${({ theme }) => theme.spacing(2)}px;
+`;
+
+internals.CloseButton = Styled(Button).attrs({ appearance: 'ghost' })``;
+
+internals.StyledTopNav = Styled(TopNavigation)`
+    background-color: ${({ theme }) => theme.palette.primary};
+`;
+
+internals.ButtonWrapper = Styled(TopNavigation)`
+    background-color: ${({ theme }) => theme.palette.slate[500]};
+`;
+
+internals.Title = Styled(Text).attrs({ category: 'h4' })`
+    color: ${({ theme }) => theme.palette.slate[100]};
+    font-size: 22px;
+    font-weight: 600;
+    font-family: 'System'
+`;
+
+internals.StyledModal = Styled(Modal)`
+    max-height: ${Dimensions.get('window').height - 10}px;
```

Comment:
Closing the loop on this (also, hi Devin 🫘 ), after a slightly more complicated detour, I switched this with the [useWindowDimensions hook](https://reactnative.dev/docs/usewindowdimensions), which automatically updates when the screen rotates, and made it an inline style so it would update on render (https://github.com/BigRoomStudios/shrexpo/pull/8/commits/753403714d6ea3e6663fc56078d885db353d3bfd)